test/cpp/microbenchmarks/filter_helpers.h

```diff
@@ -0,0 +1,371 @@
+                   grpc_call_stack* call_stack) {
+    grpc_millis deadline = GRPC_MILLIS_INF_FUTURE;
+    gpr_timespec start_time = gpr_now(GPR_CLOCK_MONOTONIC);
+    grpc_slice method = grpc_slice_from_static_string("/foo/bar");
+
+    call_args->call_stack = call_stack;
+    call_args->server_transport_data = nullptr;
+    call_args->context = nullptr;
+    call_args->path = method;
+    call_args->start_time = start_time;
+    call_args->deadline = deadline;
+    call_args->arena = gpr_arena_create(kArenaSize);
+  }
+
+  std::vector<grpc_arg> args;
+  grpc_core::ExecCtx exec_ctx;
+  FakeClientChannelFactory factory;
+};
+
+// Generic data needed for batch payloads in these microbenchmarks
+struct PayloadData {
+  grpc_metadata_batch metadata_batch_send_init;
+  grpc_metadata_batch metadata_batch_recv_init;
+  grpc_metadata_batch metadata_batch_send_trailing;
+  grpc_metadata_batch metadata_batch_recv_trailing;
+
+  gpr_atm peer_address_atm;
+
+  uint32_t recv_flags;
+
+  grpc_core::OrphanablePtr<grpc_core::ByteStream> op;
+
+  grpc_transport_stream_stats stats;
+
+  grpc_core::ManualConstructor<grpc_core::SliceBufferByteStream>
+      byte_stream_send;
+  grpc_slice_buffer slice_buffer_send;
+  grpc_slice_buffer slice_buffer_recv;
+
+  grpc_transport_stream_op_batch_payload payload;
+};
+
+// Initializes the payload such that filters won't complain when we send
+// all 6 ops down the filter stack
+void CreatePayloadForAllOps(struct PayloadData* data) {
+  grpc_transport_stream_op_batch_payload* payload = &data->payload;
+
+  memset(data, 0, sizeof(struct PayloadData));
+
+  grpc_metadata_batch_init(&data->metadata_batch_send_init);
+  grpc_metadata_batch_init(&data->metadata_batch_recv_init);
+  grpc_metadata_batch_init(&data->metadata_batch_send_trailing);
+  grpc_metadata_batch_init(&data->metadata_batch_recv_trailing);
+  payload->send_initial_metadata.send_initial_metadata =
+      &data->metadata_batch_send_init;
+  payload->send_trailing_metadata.send_trailing_metadata =
+      &data->metadata_batch_send_trailing;
+  payload->recv_initial_metadata.recv_initial_metadata =
+      &data->metadata_batch_recv_init;
+  payload->recv_trailing_metadata.recv_trailing_metadata =
+      &data->metadata_batch_recv_trailing;
+
+  payload->recv_initial_metadata.recv_flags = &data->recv_flags;
+  payload->recv_initial_metadata.peer_string = &data->peer_address_atm;
+  std::string peer_address_string = "Unknown.";
+  gpr_atm_rel_store(payload->recv_initial_metadata.peer_string,
+                    (gpr_atm)gpr_strdup(peer_address_string.data()));
+  payload->recv_message.recv_message = &data->op;
+
+  payload->collect_stats.collect_stats = &data->stats;
+
+  grpc_slice_buffer_init(&data->slice_buffer_send);
+  data->byte_stream_send.Init(&data->slice_buffer_send, 0);
+  payload->send_message.send_message.reset(data->byte_stream_send.get());
+
+  grpc_slice_buffer_init(&data->slice_buffer_recv);
+  grpc_core::SliceBufferByteStream* sbs =
```

Comment:
This is usually allocated by the transport. Could you allocated it in the dummy transport to make this more realistic? I think that should work

Reply:
Sounds good

Reply:
This is also an object that leaks, you need to Delete(...) it somewhere